server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -909,6 +910,30 @@ protected Node(
                 )
                 .collect(Collectors.toList());
 
+            Collection<Object> telemetryAwarePluginComponents = pluginsService.filterPlugins(TelemetryAwarePlugin.class)
```

Comment:
Please make sure that the `TelemetryPlugin` is not implementing `TelemetryAwarePlugin` this it would be leading to self-cycle.